searchcore/src/vespa/searchcore/proton/server/docstorevalidator.cpp

```diff
@@ -127,7 +128,7 @@ void DocStoreValidator::performRemoves(FeedHandler & feedHandler, const search::
             assert(document);
             LOG(info, "Removing document with id %s and lid %u with gid %s in bucket %s", document->getId().toString().c_str(), lid, metaData.gid.toString().c_str(), metaData.bucketId.toString().c_str());
             std::unique_ptr<RemoveOperation> remove = std::make_unique<RemoveOperation>(metaData.bucketId, metaData.timestamp, document->getId());
-            feedHandler.performOperation(FeedToken::UP(), std::move(remove));
+            feedHandler.performOperation(FeedToken(), std::move(remove));
```

Comment:
Note that this will not work properly if gid calculatiation change or document id corruption caused the validation to fail.
This is due to document meta store having the gid based upon original gid calculation with original document id, while this remove uses new gid calculation with a possibly corrupted document id.

The original validation used a batch remove feed operation that had a vector of lids to remove. I believe that operation needs to be resurrected too in some form.

Reply:
Hmm, I will look back in history and see if that is easily resurrected then.
You also had another comment on #2576. Is that still valid ?

Reply:
Plan to introduce RemoveWithGid instead.